extra_data/export.py

```diff
@@ -10,171 +10,118 @@
-
-    def start(self):
-        """Start a zmq.REP socket.
-        """
-        self._interface = REPInterface(self._context, self.port, self._buffer)
-        self._interface.daemon = True
-        self._interface.start()
+        while True:
+            events = dict(poller.poll())
 
-    def stop(self):
-        if self._interface:
-            self._interface.stop()
-            self._interface.join()
-            self._interface = None
-
-    def _serialize(self, data, metadata=None):
-        if not metadata:
-            metadata = {src: v.get('metadata', {}) for src, v in data.items()}
-
-        if self.dummy_timestamps:
-            ts = time()
-            sec, frac = str(ts).split('.')
-            frac = frac.ljust(18, '0')
-            update_dummy = {
-                'timestamp': ts,
-                'timestamp.sec': sec,
-                'timestamp.frac': frac,
-            }
-            for src in data.keys():
-                if 'timestamp' not in metadata[src]:
-                    metadata[src].update(update_dummy)
-
-        if self.protocol_version == '1.0':
-            return [self.pack(data)]
-
-        msg = []
-        for src, props in sorted(data.items()):
-            main_data = {}
-            arrays = []
-            for key, value in props.items():
-                if isinstance(value, np.ndarray):
-                    arrays.append((key, value))
-                elif isinstance(value, np.number):
-                    # Convert numpy type to native Python type
-                    main_data[key] = value.item()
-                else:
-                    main_data[key] = value
-
-            msg.extend([
-                self.pack({
-                    'source': src, 'content': 'msgpack',
-                    'metadata': metadata[src]
-                }),
-                self.pack(main_data)
-            ])
-
-            for key, array in arrays:
-                if not array.flags['C_CONTIGUOUS']:
-                    array = np.ascontiguousarray(array)
-                msg.extend([
-                    self.pack({
-                        'source': src, 'content': 'array', 'path': key,
-                        'dtype': str(array.dtype), 'shape': array.shape
-                    }),
-                    array.data,
-                ])
-
-        return msg
-
-    def feed(self, data, metadata=None):
+            if self.stopper_r in events:
+                self.stopper_r.recv()
+                break
+
+            try:
+                payload = self.buffer.get(timeout=0.1)
+            except Empty:
+                continue
+
+            if events[self.server_socket] is zmq.POLLIN:
+                msg = self.server_socket.recv()
+                if msg != b'next':
+                    print(f'Unrecognised request: {msg}')
+                    self.server_socket.send(b'Error: bad request %b' % msg)
+                    continue
+
+            self.server_socket.send_multipart(payload, copy=False)
```

Comment:
It looks like much of this class is duplicated from karabo-bridge-py. Is it practical to make more reusable alongside the serialisation code? Not as part of this PR, just a more general question.

(The answer might be that there's enough difference in the details that it's easier to keep them independent - it just looks familiar enough to be worth considering.)